src/HealthChecks/Abstractions/src/HealthReport.cs

```diff
@@ -14,23 +15,23 @@ public sealed class HealthReport
     /// <summary>
     /// Create a new <see cref="HealthReport"/> from the specified results.
     /// </summary>
-    /// <param name="entries">A <see cref="IReadOnlyDictionary{TKey, T}"/> containing the results from each health check.</param>
+    /// <param name="entries">A <see cref="IEnumerable{KeyValuePair}"/> containing the results from each health check.</param>
     /// <param name="totalDuration">A value indicating the time the health check service took to execute.</param>
-    public HealthReport(IReadOnlyDictionary<string, HealthReportEntry> entries, TimeSpan totalDuration)
+    public HealthReport(IEnumerable<KeyValuePair<string, HealthReportEntry>> entries, TimeSpan totalDuration)
```

Comment:
Changing the signature of public APIs is not allowed. We either need new overloads or some other approach.

Reply:
Introduced an _HealthReportEntryDictionary_ for use only in _DefaultHealthCheckService_. Technically not a Dictionary<TKey, TValue> but a Dictionary<TKey, List<TValue>>, but this way we won't break the existing _HealthReport_ contract.